core/capabilities/reader.go

```diff
@@ -25,6 +30,39 @@ type state struct {
 	IDsToCapabilities map[hashedCapabilityID]kcr.CapabilityRegistryCapability
 }
 
+func (r *remoteRegistryReader) LocalWorkflowDON(ctx context.Context) (capabilities.LocalDON, error) {
+	pid := r.peerWrapper.GetPeer().ID()
+	don := capabilities.DON{}
+
+	readerState, err := r.state(ctx)
+	if err != nil {
+		return capabilities.LocalDON{}, fmt.Errorf("failed to get state from registry to determine don ownership: %w", err)
+	}
+
+	dons := []capabilities.DON{}
+	for _, d := range readerState.IDsToDONs {
+		if !d.AcceptsWorkflows {
+			continue
+		}
+		for _, p := range d.NodeP2PIds {
+			if p == pid {
+				dons = append(dons, *toDONInfo(d))
+			}
+		}
+	}
+
+	if len(dons) > 0 {
```

Comment:
how would this situation arise OOI?

Reply:
It shouldn't -- @DeividasK is modifying the contract so that each node has only one WorkflowDON so I've removed the log and just fetching the first one in the list